silex_houdini/commands/export_ass.py

```diff
@@ -0,0 +1,135 @@
+            name="Info",
+            label="Info",
+            value=f"Warning : {message}",
+        )
+        label_parameter = ParameterBuffer(
+            type=str, name="label_parameter", label=f"{message}"
+        )
+
+        # Prompt the user with a label
+        prompt = await self.prompt_user(action_query, {"info": info_parameter})
+
+        return prompt["info"]
+
+    @CommandBase.conform_command()
+    async def __call__(
+        self,
+        parameters: Dict[str, Any],
+        action_query: ActionQuery,
+        logger: logging.Logger,
+    ):
+        def export_assnode(selected_object, node, out_file_name, to_return, logger):
+            node_filename = f"{out_file_name}_{node.name()}.$F4.placeholder"
+            # .placeholder because with_suffix replace .somethings in filename
+            # but we want to keep .0001 for example
+            node_filename = pathlib.Path(os.path.join(outdir, node_filename))
+            node_filename = str(
+                node_filename.with_suffix(f".{extension['short_name']}")
+            )
+            to_return.append(outdir)
+            node.parm("ar_ass_file").set(node_filename)
+            logger.info(f"Done export ass, output paths : {node_filename}")
+
+        def exec(selected_object):
+            selected_object.parm("execute").pressButton()
+
+        outdir = parameters["file_dir"]
+        outFilename = parameters["file_name"]
+
+        to_return = []
+        # create out dir if not exist
+        if not os.path.exists(outdir):
+            os.makedirs(outdir)
+
+        # get current selection
+        selected_object = [
+            item for item in hou.selectedNodes() if item.type().name() == "arnold"
+        ]
+        while len(selected_object) != 1:
+            await self._prompt_info_parameter(
+                action_query,
+                "Invalid nodes selected,\n Please select Arnold ROP nodes and continue.",
+            )
+            selected_object = [
+                item for item in hou.selectedNodes() if item.type().name() == "arnold"
+            ]
+        selected_object = selected_object[0]
+
+        # inputDependencies
+        # test current selection only composed of arnold rop nodes
+        selected_objects_types = selected_object.inputDependencies()
+        logger.info(selected_objects_types)
+
+        selected_objects_types = [
+            item[0] for item in selected_objects_types
+        ]  # [0] to avoid frames
+        logger.info(selected_objects_types)
+        not_allowed_rop = [
+            item for item in selected_objects_types if item.type().name() != "arnold"
+        ]
+        allowed_rop = [
+            item for item in selected_objects_types if item.type().name() == "arnold"
+        ]
+
+        # disable all not allowed rop nodes
+        for node in not_allowed_rop:
+            node.bypass(1)
+
+        # get extension from api
+        extension = await gazu.files.get_output_type_by_name("ass")
```

Comment:
You don't need to do that you can just put `ass` directly

Reply:
Yes but to keep same file structure in all houdini actions i think we can pass over